qcodes/instrument_drivers/stanford_research/SR86x.py

```diff
@@ -0,0 +1,494 @@
+
+        values = self._parent.visa_handle.query_binary_values("CAPTUREGET? 0,{}".format(total_size_in_kb),
+                                                              datatype='f', is_big_endian=False)
+        values = np.array(values)
+        values = values[values != 0]
+
+        data = {k: v for k, v in zip(capture_variables, values.reshape((-1, n_variables)).T)}
+
+        return data
+
+    def capture_samples(self, sample_count: int):
+        """
+        Capture a number of samples. This convenience function provides an example how we use the start and stop
+        methods. We acquire the samples by sleeping for a time and then reading the buffer.
+
+        Args:
+            sample_count (int)
+
+        Returns:
+            dict
+        """
+        capture_rate = self.capture_rate()
+        capture_time = sample_count / capture_rate
+
+        self.start_capture("CONT", "IMM")
+        time.sleep(capture_time)
+        self.stop_capture()
+
+        return self.read_capture_data(sample_count)
+
+
+class SR86x(VisaInstrument):
+    """
+    This is the code for Stanford_SR865 Lock-in Amplifier
+    """
+    _VOLT_TO_N = {1: 0, 500e-3: 1, 200e-3: 2,
+                  100e-3: 3, 50e-3: 4, 20e-3: 5,
+                  10e-3: 6, 5e-3: 7, 2e-3: 8,
+                  1e-3: 9, 500e-6: 10, 200e-6: 11,
+                  100e-6: 12, 50e-6: 13, 20e-6: 14,
+                  10e-6: 15, 5e-6: 16, 2e-6: 17,
+                  1e-6: 18, 500e-9: 19, 200e-9: 20,
+                  100e-9: 21, 50e-9: 22, 20e-9: 23,
+                  10e-9: 24, 5e-9: 25, 2e-9: 26,
+                  1e-9: 27}
+    _N_TO_VOLT = {v: k for k, v in _VOLT_TO_N.items()}
+
+    _CURR_TO_N = {1e-6: 0, 500e-9: 1, 200e-9: 2,
+                  100e-9: 3, 50e-9: 4, 20e-9: 5,
+                  10e-9: 6, 5e-9: 7, 2e-9: 8,
+                  1e-9: 9, 500e-12: 10, 200e-12: 11,
+                  100e-12: 12, 50e-12: 13, 20e-12: 14,
+                  10e-12: 15, 5e-12: 16, 2e-12: 17,
+                  1e-12: 18, 500e-15: 19, 200e-15: 20,
+                  100e-15: 21, 50e-15: 22, 20e-15: 23,
+                  10e-15: 24, 5e-15: 25, 2e-15: 26,
+                  1e-15: 27}
+    _N_TO_CURR = {v: k for k, v in _CURR_TO_N.items()}
+
+    _VOLT_ENUM = Enum(*_VOLT_TO_N.keys())
+    _CURR_ENUM = Enum(*_CURR_TO_N.keys())
+
+    _INPUT_SIGNAL_TO_N = {
+        'voltage': 0,
+        'current': 1,
+    }
+    _N_TO_INPUT_SIGNAL = {v: k for k, v in _INPUT_SIGNAL_TO_N.items()}
+
+    def __init__(self, name, address, reset=False, **kwargs):
+        super().__init__(name, address, terminator='\n', **kwargs)
+        # Reference commands
+        self.add_parameter(name='frequency',
+                           label='Frequency',
+                           unit='Hz',
+                           get_cmd='FREQ?',
+                           set_cmd='FREQ {}',
+                           get_parser=float,
+                           vals=Numbers(min_value=1e-3, max_value=2.5e6))
```

Comment:
I know this is from the old driver and you didn't write this. Never the less it seems strange to me that the max value is 2.5 MHz. As far as i can see this should be 5 KHz for the SR860 and 4 MHz for the SR865